src/dispatcher.js

```diff
@@ -197,15 +197,11 @@ var dispatcher = {
   },
   leaveOut: function(event) {
     this.out(event);
-    if (!this.contains(event.target, event.relatedTarget)) {
-      this.leave(event);
-    }
+    this.propagate(event, this.leave.bind(this), false);
   },
   enterOver: function(event) {
     this.over(event);
-    if (!this.contains(event.target, event.relatedTarget)) {
-      this.enter(event);
-    }
+    this.propagate(event, this.enter.bind(this), true);
```

Comment:
Good point. I was uncertain which way to pick. I didn't expect there to be such large differences in performance. I will update the PR.